test-galasactl-ecosystem.sh

```diff
@@ -178,6 +185,143 @@ function launch_test_on_ecosystem_with_portfolio {
+    --bootstrap ${bootstrap}"
+
+    info "Command is: $cmd"
+
+    $cmd | grep "${runname}" # Checks the RunName can be found in the output from galasactl runs get
+    rc=$?
+    # We expect a return code of '0' because the ecosystem should be able to find this test as we just ran it.
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to query the result of run ${runname} in the remote ecosystem."
+        exit 1
+    fi
+    success "Querying the result of a run in the ecosystem worked OK"
+
+    # The test above just checks that some output was found from galasactl runs get.
+    # TO DO - Get the Result of the run from the output of galasactl runs submit as well as the RunName, and make sure the Result is correct too
+    export RUN_NAME=${runname}
+}
+
+#--------------------------------------------------------------------------
+function runs_get_check_summary_format_output {
+    h2 "Performing runs get with summary format..."
+
+    run_name=$1
+
+    cd ${BASEDIR}/temp
+
+    cmd="${BASEDIR}/bin/${binary} runs get \
+    --name ${run_name} \
+    --format summary \
+    --bootstrap ${bootstrap} "
+
+    info "Command is: $cmd"
+
+    output_file="runs-get-output.txt"
+    $cmd | tee $output_file
+
+    # Check that the full test name is output
+    cat $output_file | grep "${GALASA_TEST_NAME_LONG}"
+    rc=$?
+    # We expect a return code of '0' because the test name should be output.
+    if [[ "${rc}" != "0" ]]; then 
+        error "Did not find ${GALASA_TEST_NAME_LONG} in summary output"
+        exit 1
+    fi
+
+    # Check headers
+    headers=("name" "status" "result" "test-name")
+
+    for header in "${headers[@]}"
+    do
+        cat $output_file | grep "$header"
+        rc=$?
+        # We expect a return code of '0' because the header name should be output.
+        if [[ "${rc}" != "0" ]]; then 
+            error "Did not find header $header in summary output"
+            exit 1
+        fi
+    done    
+
+    # Check that we got 2 lines out... one for the headers, on for the 1 line of test data.
+    line_count=$(cat $output_file | wc -l | xargs)
+    expected_line_count="2"
+    if [[ "${line_count}" != "${expected_line_count}" ]]; then 
+        error "line count is wrong. expected ${expected_line_count} got ${line_count}"
+        exit 1
+    fi
+
+    success "galasactl runs get --format summary seemed to work"
+}
+
+#--------------------------------------------------------------------------
+function runs_get_check_details_format_output {
+    h2 "Performing runs get with details format..."
+
+    run_name=$1
+
+    cd ${BASEDIR}/temp
+
+    cmd="${BASEDIR}/bin/${binary} runs get \
+    --name ${run_name} \
+    --format details \
+    --bootstrap ${bootstrap} "
+
+    info "Command is: $cmd"
+
+    output_file="runs-get-output.txt"
+    $cmd | tee $output_file
+
+
+    # Check that the full test name is output and formatted
+    cat $output_file | grep "test-name    :  ${GALASA_TEST_NAME_LONG}"
```

Comment:
You might like to use some wild-cards in there...
the number of spaces could be variable...
eg: grep "test-name[ \t]*:[ \t]+${GALASA_TEST_NAME_LONG}" ?